File: 0070-climbing-stairs/0070-climbing-stairs.java

class Solution {
    public int climbStairs(int n) {
        int[] dp = new int[n + 1];
        Arrays.fill(dp, -1);
        return solve(0, n, dp);
    }

    private int solve(int i, int n, int[] dp) {
        if (i == n) return 1;
        if (i > n) return 0;

        if (dp[i] != -1) return dp[i];

        dp[i] = solve(i + 1, n, dp) + solve(i + 2, n, dp);

        return dp[i];
    }
}

File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        dp = [-1] * (n+1)
        return self.helper(0, n, dp)

    def helper(self, currentLevel, n, dp):

        if currentLevel == n:
            return 1

        if currentLevel > n:
            return 0

        if dp[currentLevel] != -1:
            return dp[currentLevel]

        dp[currentLevel] = self.helper(currentLevel + 1, n, dp) + self.helper(currentLevel + 2, n, dp)

        return dp[currentLevel]